numpy/core/src/multiarray/methods.h

```diff
@@ -1,9 +1,28 @@
 #ifndef _NPY_ARRAY_METHODS_H_
 #define _NPY_ARRAY_METHODS_H_
 
+#include "npy_import.h"
+
 extern NPY_NO_EXPORT PyMethodDef array_methods[];
 
 NPY_NO_EXPORT const char *
 npy_casting_to_string(NPY_CASTING casting);
 
+/* Pathlib support */
+static inline PyObject *
+NpyPath_PathlikeToFspath(PyObject *file)
+{
+    static PyObject *os_PathLike = NULL;
+    static PyObject *os_fspath = NULL;
+    npy_cache_import("numpy.compat", "os_PathLike", &os_PathLike);
+    if (os_PathLike == NULL) {
+        return file;
```

Comment:
In this case, you should just return `NULL`. If the import failed, something is very wrong.

Reply:
Right. I guess I was still thinking in terms of the non-np.compat os.fspath and os.pathlike, which may legitimately not exist.